pkg/ebpf/events_pipeline.go

```diff
@@ -472,7 +471,7 @@ func (t *Tracee) deriveEvents(ctx context.Context, in <-chan *trace.Event) (
 					case events.PrintMemDump:
 					default:
 						// Derived events might need filtering as well
-						if !t.shouldProcessEvent(&derivative) {
+						if t.matchPolicies(&derivative) == 0 {
```

Comment:
Ok, but a question that raises now: is copying the event really necessary now that we do the matched policies copy and differentiation in between kernel and userland ? Not copying/duplicating a high frequency of derived events (like net_packet_XXX) could be good for the pipeline (and us =D).

Reply:
Good question, probably no need to copy.
Just to avoid regression surprises, let's do it (remove the copy) right after the release

Reply:
Your point makes sense and got me thinking that since we're ignoring events not set in matchPolicies() gets its bit intact for later stages of the pipeline (i.e. derivation).

So, perhaps this check of derivation can be removed:

![image](https://github.com/aquasecurity/tracee/assets/3372117/a7d81b0a-1983-4bdc-975d-6aee11a2ddbd)

I didn't check it yet and won't before get the fixes merged. As Yaniv suggested, let's move on and try those changes in the next release.

Reply:
This check of derivation is here for performance reasons - if the event doesn't have derived events and no matched policies then there is no point to push it to the next pipeline stage

Reply:
Yes, @geyslan is testing the "inner IF" need (after his fixes @josedonizetti and I think the inner IF isn't needed). He tested E2E without it and it works fine (and our E2E as well). If you know any particular corner case that still needs the inner IF:

```go
if _, ok := t.eventDerivations[eventId]; !ok {
```

let us know, but it looks like matchPolicies() takes care of everything needed now.

Reply:
But this condition won't happen. All base events are marked to the policies that have derived events as that policies needed event as well. So the base events should never have the policy bitmask zeroed and matchPolicies() will take care of everything (I believe @geyslan is testing that now or has tested it after discussion with @josedonizetti and I).

Reply:
Actually, if filters are applied to the base event, the bitmask might have been cleaned and then we would have to check the base event for derived events. So this check might be right @geyslan told me so, and I agreed. I asked for a comment there explicitly saying that (maybe giving one example).

Reply:
Consider:

A policy with:

- 1 base event (with an arg filter)
- 1 event derived from that same base (without any filter - i.e. any base sent must arrive at the printer)

The base (first to be checked) policy has a unmatch, so it has the policy bit cleared.

When returning from `matchPolicies()`, as it is just one policy, this return has the bit cleared and it's 0, so if we don't check, as a requirement, that this event has derivatives from it, the event that should derivate goes to /dev/null.